### ISSUE TITLE
TSV: state who to handle special characters in strings

### ISSUE THREAD

**Issue by Tpt** [w3c/sparql-results-csv-tsv#10]
The specification does not explicitely states how quotes and ASCII control characters (`\0`...) should be escaped. I might be nice to
add some sentences about it.

A note to state that the `"` quote should be prefered to the `'` quote might also be nice to get some kind of "canonical" TSV serialization.

**Comment by TallTed** [comment]
> A note to state that the " quote should be prefered to the ' quote might also be nice to get some kind of "canonical" TSV serialization.

This preference is commonly dictated by the data. If my data has lots of `"` characters and few or no `'`, I'd prefer to use the `'` to quote each field, minimizing the need for inline escapes.

Putting some guidance like yours into a distinct "notes on canonicalization" section would probably be OK.

**Comment by domel** [comment]
I think that it is bad idea to change / overwrite the basic spec that is used. TSV has no official spec, but CSV has. And in that spec there is no information about `'`. It recommends to use double quotes (or nothing).

**Comment by afs** [comment]
TSV does have an official spec!
https://www.iana.org/assignments/media-types/text/tab-separated-values

**Comment by afs** [comment]
In TSV, the quotes and escapes are from the RDF term writing.

https://w3c.github.io/sparql-results-csv-tsv/spec/index.html#tsv-terms
"by using the syntax that SPARQL and Turtle use."

From what I see on the web, in Turtle, `"` is more common.

Each needs escaping checking (`'` in names seems to catch data writing system out).

Some advice-text would be useful - less than formal, single-choice canonicalization.

**Comment by domel** [comment]
> TSV does have an official spec! https://www.iana.org/assignments/media-types/text/tab-separated-values

Yes and no. It's rather a documentation for media type than official spec (that is RFC or STD). Regardless of the naming, there is nothing about `'`.

**Comment by afs** [comment]
> And in that spec there is no information about `'`. It recommends to use double quotes (or nothing).

This issue (#10) is specific to TSV. For CSV, we should, of course, use `"`.

**Comment by afs** [comment]
This "needs discussion" issue was discussed during the telecon of 2023-11-30.

From the issue thread above, are we agreed that:

1. TSV does not make any special case of `"` or `'` because it is separation by a raw TAB.
1. Turtle serializers more commonly use `"`.
1. The current spec text covers quoting and control characters "by using the syntax that SPARQL and Turtle use." ([section 5.1](https://www.w3.org/TR/sparql12-results-csv-tsv/#tsv-terms)). Hence, no raw TABs in RDF term text.
1. The text would benefit from expanding, such as having inline examples.

Anything else?

**Comment by afs** [comment]
Related to handling characters: the [TSV Media Type](https://www.iana.org/assignments/media-types/text/tab-separated-values) does not specify the character set. Nowadays, the "default" for "text/" is UTF-8, a change from the original ASCII.

We can mention this and suggest ("SHOULD") that no character set is treated as UTF-8.
